opp_path_selection/include/opp_path_selection/filter_impl.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2019 Southwest Research Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OPP_PATH_SELECTION_FILTER_IMPL_H
+#define OPP_PATH_SELECTION_FILTER_IMPL_H
+
+#include "opp_path_selection/filter.h"
```

Comment:
This technically introduces a circular dependency since your header includes this file. This is handled by the header guards, but isn't necessarily great. The above comment would address this

Reply:
We should fix both packages as you suggest. opp_area_selection and opp_path_selection.